internal/host/host.go

```diff
@@ -1050,11 +1050,14 @@ func (m *Manager) reportValidationStatusChanged(ctx context.Context, vc *validat
 					m.metricApi.HostValidationChanged(models.HostValidationID(v.ID))
 					eventgen.SendHostValidationFailedEvent(ctx, m.eventsHandler, *h.ID, h.InfraEnvID, h.ClusterID,
 						hostutil.GetHostnameForMsg(h), v.ID.String())
-				}
-				if v.Status == ValidationSuccess && currentStatus == ValidationFailure {
+				} else if v.Status == ValidationSuccess && currentStatus == ValidationFailure {
```

Comment:
```suggestion
				} else if v.Status == ValidationSuccess || v.Status == ValidationPending && currentStatus == ValidationFailure {
```

Reply:
there are nicer ways to do this but you get the idea

Reply:
Unsure I understand what you suggested here.
This `else if` handles the case of a validation changing from success to failure.

Reply:
OK, this comment was ment for the validation failed section.
You are right we should send an event for every validation that failed (and wasn't failing before) e.g.
```
if v.Status == ValidationFailure && currentStatus != ValidationFailure {
```

Although it seems that we don't send events for failing validation (unless they move from success to failed) I'd like to look into it and fix it in another PR.

Reply:
https://github.com/openshift/assisted-service/pull/4226